File: _BitManipulation/201. Bitwise AND of Numbers Range.py
import operator
from functools import reduce

class Solution:
    def rangeBitwiseAnd(self, m, n):
        fac = 0

        while(m!=n):
            m >>= 1
            n >>= 1
            fac+=1

        return (m&n)*2**fac

if __name__ == "__main__":

    m, n = 108, 110
    obj = Solution()
    print(obj.rangeBitwiseAnd(m,n))

File: _BitManipulation/cpp/201_Bitwise_AND_of_Numbers_Range.cpp
#include <iostream>
#include <cmath>
using namespace std;
class Solution {
public:
    int rangeBitwiseAnd(int m, int n) {
        double fac = 0;
        
        while(m!=n){
            m >>= 1;
            n >>= 1;
            fac+=1;
        }

        return (m&n)*pow(2,fac);
    }
};

int main(){
    int m = 0;
    int n = 2147483647;
    
    Solution obj;
    cout << obj.rangeBitwiseAnd(m,n);
    return 0;
}